pinot-common/src/main/java/org/apache/pinot/common/data/objects/MapObject.java

```diff
@@ -0,0 +1,48 @@
+package org.apache.pinot.common.data.objects;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.pinot.common.data.PinotObject;
+
+import com.beust.jcommander.internal.Lists;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class MapObject implements PinotObject {
+
+  private static ObjectMapper _MAPPER = new ObjectMapper();
+  Map<String, Object> _stringMap;
```

Comment:
Would using generics help to support typed maps?

Reply:
PinotMapObject?